src/framework/theme/component/modal/modal.component.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import {
+  Modal,
+  View,
+  StyleSheet,
+  TouchableWithoutFeedback,
+} from 'react-native';
+
+interface ModalComponentProps {
+  visible: boolean;
+  component: JSX.Element | null;
+  isBackDropAllowed: boolean;
+  onBackdrop: () => void;
+}
+
+export class ModalComponent extends React.Component<ModalComponentProps> {
+
+  renderBackground(): React.ReactNode {
+    return <View style={styles.background}/>;
+  }
+
+  renderComponent(): React.ReactNode {
```

Comment:
Split this to 2 methods

Reply:
It even could be conditionally wrapped to `Touchable` instead of creating the same element as passed child